hs_communities/static/js/dual-listbox.js

```diff
@@ -0,0 +1,177 @@
+var TitleAssistantApp = new Vue({
+    el: "#title-assistant",
+    data: {
+        title: '',
+        regionSelected: '',
+        subtopic: '',
+        startYear: '',
+        endYear: '',
+        yearOngoing: '',
+        location: '',
+        errmsg: '',  // Inline feedback for the user in red
+        topics: {
+            itemsList: [], // hold the master list of topics relayed by the HTML template
+            unselectedItems: [],
+            selectedItems: [],
+            selectedValues: ''
+        },
+        startYearParen: '',  // Paren items are for special display, requested by the customer to appear that way
+        endYearParen: ''
+    },
+    methods: {
+        evalParenFields: function () {  // Parenthetical fields are specified by customer to be surrounded by parentheses
+            if (this.$data.startYear) {
+                this.$data.startYearParen = "(" + this.$data.startYear
+            } else {
+                this.$data.startYearParen = ""
+            }
+            if (this.$data.endYear) {
+                this.$data.endYearParen = this.$data.endYear + ")"
+            } else {
+                this.$data.endYearParen = ""
+            }
+        },
+        itemMoved: function () {  // User has selected some topics so concat them with commas for display
+            this.$data.topics.selectedValues = this.$data.topics.selectedItems.map(x => x.value).join(', ');
+            this.updateTitle()
+        },
+        checkOngoing: function () {  // Checking Ongoing box indicates End Year contains the string Ongoing
+            this.$data.endYear = this.$data.yearOngoing;
+            this.updateTitle()
+        },
+        updateRegion: function () {
+            this.updateTitle(items.join(','))
+        },
+        updateEndDate: function () {
+            $("#end-date-ongoing").prop("checked", false);
+            this.updateDate()
+        },
+        updateDate: function () {
+            if (isValidYear(this.$data.startYear) && isValidYear(this.$data.endYear)) {
+                this.$data.errmsg = "";
+                this.updateTitle()
```

Comment:
@Maurier Note here and many other places, the use of `this.$data` I didn't have time to search around for posts on this approach, versus `this.errmsg` for example (without $data) Please let me know if you have an opinion on this.

Reply:
Both ways to access these data inside the Vue instance will work, so I'm fine with it. The `$data` object is also useful if we need to access these data outside of the instance.